Sources/SmokeDynamoDB/AWSDynamoDBClientConfiguration.swift

```diff
@@ -0,0 +1,195 @@
+        self.contentType = contentType
+        self.clientDelegate = JSONAWSHttpClientDelegate<DynamoDBError>(requiresTLS: useTLS)
+        self.connectionPoolConfiguration = connectionPoolConfiguration
+        self.awsRegion = awsRegion
+        self.retryConfiguration = retryConfiguration
+        self.eventLoopGroup = AWSClientHelper.getEventLoop(eventLoopGroupProvider: eventLoopProvider)
+        self.timeoutConfiguration = timeoutConfiguration
+        self.reportingConfiguration = reportingConfiguration
+                
+        self.reportingProvider = { (logger, internalRequestId, eventLoop) in
+            return StandardHTTPClientCoreInvocationReporting(
+                logger: logger,
+                internalRequestId: internalRequestId,
+                traceContext: traceContext,
+                eventLoop: eventLoop)
+        }
+    }
+    
+    public init(
+        credentialsProvider: CredentialsProvider,
+        awsRegion: AWSRegion,
+        endpointHostName endpointHostNameOptional: String? = nil,
+        endpointPort: Int = 443,
+        requiresTLS: Bool? = nil,
+        service: String = "dynamodb",
+        contentType: String = "application/x-amz-json-1.0",
+        target: String? = "DynamoDB_20120810",
+        timeoutConfiguration: HTTPClient.Configuration.Timeout = .init(),
+        retryConfiguration: HTTPClientRetryConfiguration = .default,
+        eventLoopProvider: HTTPClient.EventLoopGroupProvider = .createNew,
+        reportingConfiguration: SmokeAWSClientReportingConfiguration<DynamoDBModelOperations>
+            = SmokeAWSClientReportingConfiguration<DynamoDBModelOperations>(),
+        connectionPoolConfiguration: HTTPClient.Configuration.ConnectionPool? = nil)
+    where InvocationReportingType == StandardHTTPClientCoreInvocationReporting<AWSClientInvocationTraceContext> {
+        self.init(credentialsProvider: credentialsProvider,
+                  awsRegion: awsRegion,
+                  endpointHostName: endpointHostNameOptional,
+                  endpointPort: endpointPort,
+                  requiresTLS: requiresTLS,
+                  service: service,
+                  contentType: contentType,
+                  target: target,
+                  traceContext: AWSClientInvocationTraceContext(),
+                  timeoutConfiguration: timeoutConfiguration,
+                  retryConfiguration: retryConfiguration,
+                  eventLoopProvider: eventLoopProvider,
+                  reportingConfiguration: reportingConfiguration,
+                  connectionPoolConfiguration: connectionPoolConfiguration)
+    }
+    
+    public func createOperationsClient(forTableName tableName: String)
+    -> AWSGenericDynamoDBTableOperationsClient<InvocationReportingType> {
+        return AWSGenericDynamoDBTableOperationsClient(config: self,
+                                                       tableName: tableName)
+    }
+    
+    internal func createClient() -> HTTPOperationsClient {
```

Comment:
Maybe rename to `createHTTPClient` to distinguish from the operations client and the AWS client.

Reply:
Changed to `createHTTPOperationsClient`